src/sst/core/statapi/statlghistogram.h

```diff
@@ -0,0 +1,382 @@
+        SST_SER(m_numBins);
+        SST_SER(m_OOBMinCount);
+        SST_SER(m_OOBMaxCount);
+        SST_SER(m_itemsBinnedCount);
+        SST_SER(m_totalSummed);
+        SST_SER(m_totalSummedSqr);
+        SST_SER(m_binsMap);
+        SST_SER(m_dumpBinsOnOutput);
+        SST_SER(m_includeOutOfBounds);
+        // SST_SER(m_Fields); // Rebuilt by stat output object
+    }
+
+protected:
+    /**
+        Adds a new value to the histogram. The correct bin is identified and then incremented. If no bin can be found
+        to hold the value then a new bin is created.
+    */
+    void addData_impl_Ntimes(uint64_t const N, BinDataType const value) override
+    {
+        // Check to see if the value is above or below the min/max values
+        if ( value < getBinsMinValue() ) {
+            m_OOBMinCount += N;
+            return;
+        }
+        if ( value > getBinsMaxValue() ) {
+            m_OOBMaxCount += N;
+            return;
+        }
+
+        // This value is to be binned...
+        // Add the "in limits" value to the total summation's
+        m_totalSummed += static_cast<BinDataType>(N) * value;
+        m_totalSummedSqr += static_cast<BinDataType>(N) * (value * value);
+
+        // Increment the Binned count (note this <= to the Statistics added Item Count)
+        ++m_itemsBinnedCount;
+
+        // Figure out what the starting bin is and find it in the map
+        // To support signed and unsigned values along with floating point types,
+        // the calculation to find the bin_start value must be done in floating point
+        // then converted to BinDataType
+        double const      log2_value = std::log2(static_cast<double>(value));
+        double const      bin_floor_value = std::floor(log2_value); // Find the floor of the value
+        BinDataType const bin_start = static_cast<BinDataType>(bin_floor_value);
+        //      printf("DEBUG: value = %d, junk1 = %f, calc2 = %f, calc3 = %f : bin_start = %d, item count = %ld, \n",
+        //      value, calc1, calc2, calc3, bin_start, getStatCollectionCount());
+
+        HistoMapItr_t bin_itr = m_binsMap.find(bin_start);
+
+        // Was the bin found?
+        if ( bin_itr == m_binsMap.end() ) {
+            // No, Create the bin and set a value of 1 to it
+            m_binsMap.insert(std::pair<BinDataType, CountType>(bin_start, static_cast<CountType>(N)));
+        }
+        else {
+            // Yes, Increment the specific bin's count
+            bin_itr->second += static_cast<CountType>(N);
+        }
+    }
+
+    void addData_impl(BinDataType const value) override { addData_impl_Ntimes(1, value); }
+
+private:
+    /** Count how many bins are active in this histogram */
+    NumBinsType getActiveBinCount() { return m_binsMap.size(); }
+
+    /** Count how many bins are available */
+    NumBinsType getNumBins() { return m_numBins; }
+
+    /** Get the width of a bin in this histogram */
+    NumBinsType getBinWidth() { return m_binWidth; }
```

Comment:
Getters should usually be `const`.

i.e.

```c++
    /** Count how many bins are active in this histogram */
    NumBinsType getActiveBinCount() const { return m_binsMap.size(); }

    /** Count how many bins are available */
    NumBinsType getNumBins() const { return m_numBins; }

    /** Get the width of a bin in this histogram */
    NumBinsType getBinWidth() const { return m_binWidth; }
```
Also maybe `getBinCountByBinStart` which follows.